target/linux/ramips/dts/mt7628an_xiaomi_miwifi-nano.dts

```diff
@@ -67,9 +66,13 @@
 	status = "disabled";
 };
 
+&esw {
+	mediatek,portmap = <0x2f>;
+	mediatek,portdisable = <0x2a>;
+};
+
 &wmac {
 	status = "okay";
-	ralink,mtd-eeprom = <&factory 0x4>;
```

Comment:
This is not "mtd-mac-address", but "mtd-eeprom"!
So, is this really ignored? I calculated MAC addresses from that (using 0x8) which would be wrong then, too.

Reply:
It is added by b28e94d4bfa1. I'm not sure about the author's intention...
Anyway, ralink,* is only used in rt2x00, and mt76 only uses mediatek,*.
```
$ grep -rIn "ralink,"
drivers/net/wireless/ralink/rt2x00/rt2x00dev.c:1022:    if (!of_property_read_u32(np, "ralink,2ghz",
drivers/net/wireless/ralink/rt2x00/rt2x00dev.c:1025:    if (!of_property_read_u32(np, "ralink,5ghz",
drivers/net/wireless/ralink/rt2x00/rt2800soc.c:227:     { .compatible = "ralink,rt2880-wmac" },
drivers/net/wireless/ralink/rt2x00/rt2x00eeprom.c:49:   list = of_get_property(np, "ralink,mtd-eeprom", &size);
drivers/net/wireless/ralink/rt2x00/rt2x00eeprom.c:82:   if (of_find_property(np, "ralink,mtd-eeprom-swap", NULL))
drivers/net/wireless/ralink/rt2x00/rt2x00eeprom.c:109:  if (np && of_property_read_string(np, "ralink,eeprom", &eep) == 0)
drivers/net/wireless/ralink/rt2x00/rt2800lib.c:11183:           if (!of_property_read_u32(np, "ralink,led-polarity",

$ grep -rIn "mediatek,"
drivers/net/wireless/mediatek/mt76/mt7603/soc.c:67:     { .compatible = "mediatek,mt7628-wmac" },
drivers/net/wireless/mediatek/mt76/eeprom.c:29:         list = of_get_property(np, "mediatek,mtd-eeprom", &size);
```

Reply:
Ah, thanks for the research. Indeed, the commit message talks of MAC addresses. Do you own the device, and may just quickly check whether the address is in 0x4 or 0x8?

Reply:
It's in 0x4.
`0040000  28 76 00 02 f0 b4 29 xx  xx xx 00 00 00 00 00 00  |(v....).........|`

Reply:
Okay, so you are right then and I will look for the similar cases I remember.

Reply:
https://patchwork.ozlabs.org/patch/1214746/
This will remove all cases (seems like quite a lot of copy/paste happened). Thanks for finally solving this odd thing.
I'd actually remove this change from the patchset here, as with my patch we have a separate commint as well as a more extensive explanation and your patch provides enough other valuable changes anyway.

Reply:
Great! I've never noticed that... Thanks for your work. :)

Reply:
I actually stumbled over this when changing label_mac in 02_network from (eth0+X) to flash locations in the first place (setting those strange 0x8 offsets). After all, even if nobody uses the label MAC address, having implemented it revealed so many errors in the existing setup that it was worth it just for that :-)